test/e2e/samples.go

```diff
@@ -0,0 +1,132 @@
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	operator "github.com/redhat-developer/build/pkg/apis/build/v1alpha1"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+)
+
+// SamplesTesting encapsulate the probes executed against sample files in build-operator.
+type SamplesTesting struct {
+	t   *testing.T           // testing instance
+	ctx *framework.TestCtx   // operator-sdk test context
+	f   *framework.Framework // operator-sdk test framework parameters
+	ns  string               // namespace
+}
+
+// SampleFiles group resources employed during each test interaction
+type SampleFiles struct {
+	ClusterBuildStrategy string // ClusterBuildStrategy file path
+	BuildStrategy        string // BuildStrategy file path
+	Build                string // Build file path
+	BuildRun             string // BuildRun  file path
+}
+
+// SecretName kubernetes secret resource name
+const SecretName = "e2e-registry"
+
+// amendOutputImage patch Build resource to include reference to container registry credentials
+// secret.
+func (s *SamplesTesting) amendOutputImage(b *operator.Build, imageRepo, repoSecret string) {
+	if imageRepo == "" || repoSecret == "" {
+		return
+	}
+	imageURL := fmt.Sprintf("%s:%s", imageRepo, b.Name)
```

Comment:
`imageRepo` seems to be coming from `os.Getenv(EnvVarImageRepo)` , generating a new variable `imageURL` with a modified name seems to be misleading, can you double check this please? . The reasoning is as follows:
-  If I as a user specified the `TEST_IMAGE_REPO`, that already contains the host and application name. The application name is not relevant, while all you want to test is that you can push, so you can be always pushing to the same app name.

Reply:
@qu1queee I like this change. Having #117 in mind, we will need to use different image URLs anyway as with parallel build, two concurrent instance pushing to the same URL would fail.

Reply:
@SaschaSchwarze0 you are right! yes, thinking on the parallel e2e make sense. @otaviof maybe just to make sure this is very clear on the docs?

Reply:
I've just updated the documentation. Thanks.